src/components/ProgressBar/ProgressBar.stories.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useRef, useState } from "@storybook/addons"
+import { ComponentStory } from "@storybook/react"
+import { FC } from "react"
+import { Button } from "../Button"
+import { ProgressBar } from "./ProgressBar"
+
+/**
+ * See Storybook Docs: Writing Stories
+ * https://storybook.js.org/docs/basics/writing-stories/
+ */
+
+export default {
+  title: "Components/ProgressBar",
+}
+
+const IndeterminateTemplate: ComponentStory<typeof ProgressBar> = (args) => (
+  <ProgressBar {...args} />
+)
+
+export const Indeterminate = IndeterminateTemplate.bind({})
+
+const DeterminateTemplate: ComponentStory<FC<{ interval: number }>> = (
```

Comment:
This is a great visual for a "real-world" scenario. I'd recommend also adding a story of a standalone determinate instance with Storybook controls, just so a dev can check that `50` is actually halfway.